docs/auth/README.adoc

```diff
@@ -0,0 +1,112 @@
+
+== Usage
+
+To use the Auth SDK you'll first need to:
+
+* Have a Keycloak instance. See the <<Setting up Keycloak>> section.
+* A configuration file added to the app directory. See the <<Configuration File>> section.
+
+=== Setting up Keycloak
+
+==== Using OpenShift
+
+* If you do not have mobile services enabled in your openshift cluster follow this link:https://github.com/aerogear/mobile-core/blob/master/docs/walkthroughs/local-setup.adoc[Local Setup] guide.
+* Navigate to your Openshift cluster and in the Service Catalog search for the Keycloak service.
+* Click on the Keycloak service and you will be prompted to fill in details about your app.  For now you can leave these as they are.  Navigate through the setup and click Create.
+This will provision the Keycloak service in the project you specified and create a public Client to be used with an app along with a bearer-only client.
+See link:http://www.keycloak.org/docs/latest/server_admin/index.html#oidc-clients[Keycloak OIDC client documentation].
+
+==== Specifying the Redirect URI
+For Cordova based applications, localhost can be specified as a redirect URI in the Keycloak client when using an in-app browser:
+
+ * `http://localhost/*`
+
+=== Adding Auth Package
+Add the Auth package with NPM:
+----
+npm install aerogear/auth@latest --save
+----
+
+=== Configuration File
+
+A `mobile-services.json` file must exist in the app directory. It should specify configuration
+for Keycloak. This configuration can be generated by the link:https://github.com/aerogear/mobile-cli[AeroGear Mobile CLI].
+
+For an example of Keycloak configuration see the example apps link:../../example/cordova/www/mobile-services.json[mobile-services.json].
+
+The Auth SDK will use this configuration to communicate with Keycloak.
+
+=== Auth Service Initialisation
+The `AuthService` can be retrieved by importing it. The `mobile-services.json` file must also be referenced as it contains the Keycloak configuration.
+----
+import { AuthService } from 'aerogear/auth';
+let mobileServiceConfig = require("../mobile-services.json");
+----
+Before the `AuthService` can be used, the `init` function must be invoked once in an app.
+
+----
+let authService = new AuthService(mobileServiceConfig);
+
+let initOptions = {onLoad: "login-required"};
+authService.init(initOptions)
+    .then(() => {
+        // successful init & authentication
+    })
+    .catch((err) => {
+        // initialisation error
+    });
+----
+
+You can pass `login-required` or `check-sso` to the init function. `login-required` will authenticate the client if the user is logged-in to Keycloak or display the login page if not.
+`check-sso` will only authenticate the client if the user is already logged-in, if the user is not logged-in the browser will be redirected back to the application and remain unauthenticated.
+
+To enable either of these, just pass the config to the init method:
+
+```
+authService.init({onLoad: "check-sso"})
```

Comment:
No longer needed to pass "check-sso". We are doing that on the library level. If you feel that's wrong we going to remove it from the code.